src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/Package.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.CommandLine.Parsing;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Commands.Package.Update
+{
+    internal record Package
+    {
+        public required string Id { get; init; }
+        public required VersionRange? VersionRange { get; init; }
+
+        internal static List<Package> Parse(ArgumentResult result)
+        {
+            if (result.Tokens.Count == 0)
+            {
+                return [];
```

Comment:
I'm not sure but was thinking that this would allocate an array. Would `Enumerable.Empty<Package>()` work here and reduce allocations?

Reply:
The return type in this iteration of the PR is `List<Package>`, so `Enumerable.Empty<Package>()` is not assignable to that.  But by changing the method return type to `IReadOnlyList<Package>`, that makes `return []` return a static empty value.